tests/testsuite/precise_pre_release.rs

```diff
@@ -65,11 +65,12 @@ fn update_pre_release() {
     p.cargo("update my-dependency --precise 0.1.2-pre.0 -Zunstable-options")
         .masquerade_as_nightly_cargo(&["precise-pre-release"])
         .with_stderr_data(str![[r#"
+[UPGRADING] my-dependency ^0.1.1 -> ^0.1.2-pre.0
 [UPDATING] `dummy-registry` index
-[UPDATING] my-dependency v0.1.1 -> v0.1.2-pre.0
 
 "#]])
```

Comment:
This seems to incorrectly be interacting with the precise-prerelease feature.  The intention of that feature is that  we should occasionally allow matching pre-releases with a regular version requirement.